src/gfn/gflownet/base.py

```diff
@@ -71,17 +71,12 @@ def to_training_samples(self, trajectories: Trajectories) -> TrainingSampleType:
         """Converts trajectories to training samples. The type depends on the GFlowNet."""
 
     @abstractmethod
-    def loss(self, env: Env, training_objects: Any):
+    def loss(self, env: Env, training_objects: Any, **kwargs: Any):
```

Comment:
Why adding kwargs instead of `recalculate_all_logprobs`?

Reply:
not needed indeed